kafka/informationModel.go

```diff
@@ -0,0 +1,43 @@
+package kafka
+
+import (
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/confluentinc/confluent-kafka-go/v2/schemaregistry/serde/avro"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Informasjonsforvaltning/fdk-resource-service/config/logger"
+	modelAvro "github.com/Informasjonsforvaltning/fdk-resource-service/model/avro"
+)
+
+type InfoModelInputType interface {
+	InfoModelEvent() (modelAvro.InformationModelEvent, error)
+}
+
+type InfoModelInput struct {
+	message      *kafka.Message
+	deserializer *avro.SpecificDeserializer
+}
+
+func (input InfoModelInput) InfoModelEvent() (modelAvro.InformationModelEvent, error) {
+	event := modelAvro.InformationModelEvent{}
+	err := input.deserializer.DeserializeInto(*input.message.TopicPartition.Topic, input.message.Value, &event)
+	return event, err
+}
+
+func ConsumeInfoModelMessage(input InfoModelInput) error {
```

Comment:
Her bruker du Input type og i dataset bruker du InputType. Jeg synes disse typene virker litt forvirrende.

Reply:
`*InputType` er interface og `*Input` er struct basert på samme interface, begge deler vil derfor være lov å bruke som input her. Men ja, forvirrende. Dunno hva go-gurus villle sagt, men føler å bruke struct gir mest mening, bytter på datasett.

Reply:
min lokale go-guru, dvs testene mine, sa "Nei! Fy!". Fikset denne til å bruke `InfoModelInputType`

Fungerer her siden jeg ikke har implementert db osv for informasjonsmodeller